### ISSUE TITLE
support releases of ocw-course-hugo-theme

### ISSUE THREAD

**Issue by gumaerc** [mitodl/release-script#293]
As a developer, I'd like to be able to have a release pipeline for managing releases of [`ocw-course-hugo-theme`](https://github.com/mitodl/ocw-course-hugo-theme)

#### Acceptance Criteria:

- [ ] In a `#ocw-course-hugo-theme` slack channel, I can run `@doof release notes` and get a list of recent PR's merged in that repo with a button to start a new release candidate
- [ ] When the changes are merged into `release-candidate`, the pseudo-version should be obtained based on that branch and an altered `go.mod` file should be pushed to the `release-candidate` branch of `ocw-course-hugo-starter`

To clarify on this requirement, the expected pseudo-version of `ocw-course-hugo-theme` can be obtained with:
```
TZ=UTC git --no-pager show   --quiet   --abbrev=12   --date='format-local:%Y%m%d%H%M%S'   --format="%cd-%h"
```
This should give you something like:
```
c4103@c4103-pc:~/Code/ocw-course-hugo-theme$ TZ=UTC git --no-pager show   --quiet   --abbrev=12   --date='format-local:%Y%m%d%H%M%S'   --format="%cd-%h"
20210108222117-21453525a126
```
This string should be injected into the go.mod file as follows:
```
module github.com/mitodl/ocw-course-hugo-starter

go 1.13

require github.com/mitodl/ocw-course-hugo-theme v0.0.0-$OCW_COURSE_HUGO_THEME_PSEUDO_VERSION // indirect

```
The file should then be committed to the `release-candidate` branch of `ocw-course-hugo-starter` to ensure that the RC build pulls in the proper version of the theme.

 - [ ] Once the release is approved and completed, the same process should be done on the `release` branch of `ocw-course-hugo-starter`

#### Out of Scope

- I've been leaving the version number as the default v0.0.0 for now since we haven't had a release yet, but if we can set the release version here that would be good, might be out of scope though.

**Comment by pdpinch** [comment]
How/is this different from what we does with PyPI releases?

**Comment by noisecapella** [comment]
Every time there is a new release for `ocw-course-hugo-theme`, we need to update the referenced version in `ocw-course-hugo-starter`'s `go.mod` file. When the release for `ocw-course-hugo-theme` is finished, this would create a pull request in `ocw-course-hugo-starter` with the git hash of that release. Then we can do a release for `ocw-course-hugo-starter`.

**Comment by noisecapella** [comment]
FYI I think I'm going to have it just commit the `go.mod` change to `main` instead of creating an extra pull request for it. We will still have a chance to review that code when `ocw-course-hugo-starter` goes through its own release process
